Added Sharp LC-20SH1E

RobThree opened the pull request:
I'm not quite sure if this needs to be `Sharp` or `Sharp{1}`; most buttons detect both ways, sometimes it's the first, sometimes the latter. However, the press duration doesn't seem to always matter...

Remote: `GA339WJSA`

![GA339WJSA](https://github.com/user-attachments/assets/e06c8e10-f390-41cf-86c1-65a5c18f0d0b)

bengtmartensson reviewed (changes requested):
Lines 30, 31, 33 are not conforming; contain the wrong number of fields.

bengtmartensson commented:
Thank you for the contribution.

Just an aside: "offical" CSV allows commas in the field, IF the field is enclosed in quotes. IrScrutinizer groks this, but I am not sure if there are program around that do not.

RobThree commented:
No, you're right, I think it's better to play on the safe side and not allow commas.

That said, I just added a testsignal in Irscrutinizer, added a name and exported and got this:

```csv
functionname,protocol,device,subdevice,function
foo, bar,NEC1,12,34,56
```
(That's how this issue happened in the first place).

So maybe it's good to have IrScrutinizer warn (or even flatout not allow) commas (and other things like [non-ASCII codes](https://github.com/probonopd/irdb/pull/188#pullrequestreview-2681300856)) on these kind of issues?

bengtmartensson commented:
> So maybe it's good to have IrScrutinizer warn (or even flatout not allow) commas (and other things like https://github.com/probonopd/irdb/pull/188#pullrequestreview-2681300856) on these kind of issues?

That should go into the [export format](https://github.com/bengtmartensson/IrScrutinizer/blob/master/src/main/config/exportformats.d/Irdb_csv.xml). Care to  submit a PR?

Franky speaking, I am sick and tired of this silly file format and its restrictions, and I have said it several times see [the discussions](https://github.com/probonopd/irdb/discussions/categories/general). BBD (broken by design).

RobThree commented:
> That should go into the [export format](https://github.com/bengtmartensson/IrScrutinizer/blob/master/src/main/config/exportformats.d/Irdb_csv.xml). Care to submit a PR?

I'm not much of a Java developer and I don't have the/an environment set up for it. It's not that I'm not willing, I may in the future. Just don't hold your breath 😅

(As an aside: I **have** been playing around implementing a C#/.Net GIRS client for the [Arduino device](http://www.harctoolbox.org/arduino_nano.html) but I ran into some issues (top of my head: some things were unclear in the documentation I think I remember) and the project has been gathering some dust since...)

On the matter of CSV: I agree that it is simplistic, but that also makes it easy for people to build software around it. In basis I don't think it's a bad choice, but the problems you highlight in linked discussion are, indeed, hard - if not impossible - to solve in CSV. And when that becomes an issue of holding the project back then it's time to look for alternatives.

As for me: I play with this off-and-on at times. I do stumble into things, some things are unclear, sometimes I just miss or overlook something and sometimes the scattered information is the issue. Also - and I don't mean to be mean - I think some of the documentation (mostly on harctoolbox.org) assumes some level of prior knowledge or assumes some familiarity with the topic and, for me, that was what made it difficult to get started. I do think I've got the hang of it in the sense that I better understand what's going on, how the CSV's and this IRDB project works etc. but in all honesty it's just a lot and sometimes I just want to get sh*t done and move on 😜  so at times I may miss details. I'm trying my best to be a good contributor though,

bengtmartensson commented:
> I'm not much of a Java developer and I don't have the/an environment set up for it 

Export formats are XSLT 1.0. Java not required. [Dokumentation of export formats](http://www.harctoolbox.org/IrScrutinizer.html#Adding+new+export+formats)

> I think some of the documentation (mostly on harctoolbox.org) assumes some level of prior knowledge or assumes some familiarity with the topic and, for me, that was what made it difficult to get started.

That is the difference between documentation and tutorials. 

You are on your way to Enlightenment, Rob!

RobThree commented:
> Export formats are XSLT 1.0. Java not required. [Dokumentation of export formats](http://www.harctoolbox.org/IrScrutinizer.html#Adding+new+export+formats)

Well, yes, but I think the better option would be to have IrScrutinizer point out the mistakes (maybe highlight the row in red or make the text in the cell with the mistake red or - for the colorblind - maybe add a ⚠️ in the cell or...).

> That is the difference between documentation and tutorials.
> 
> You are on your way to Enlightenment, Rob!

😅 Thanks!

bengtmartensson commented:
> I think the better option would be to have IrScrutinizer point out the mistakes (maybe highlight the row in red or make the text in the cell with the mistake red or - for the colorblind - maybe add a ⚠️ in the cell or...).

Note that this would be dependent on the export format (most export format does not prohibit commas for example). Then the export fomat must contain machine readable requirements for the command names.  Cf https://github.com/bengtmartensson/IrScrutinizer/issues/87, https://github.com/bengtmartensson/IrScrutinizer/issues/145.
